File: Assets/Bundles/Client/Resources/TS/Controllers/Global/Damage/Projectile/ProjectileCollideClientSignal.lua
-- Compiled with unity-ts v2.1.0-75
local ProjectileCollideClientSignal
do
	ProjectileCollideClientSignal = setmetatable({}, {
		__tostring = function()
			return "ProjectileCollideClientSignal"
		end,
	})
	ProjectileCollideClientSignal.__index = ProjectileCollideClientSignal
	function ProjectileCollideClientSignal.new(...)
		local self = setmetatable({}, ProjectileCollideClientSignal)
		return self:constructor(...) or self
	end
	function ProjectileCollideClientSignal:constructor(projectile, hitPosition, normal, velocity, hitEntity)
		self.projectile = projectile
		self.hitPosition = hitPosition
		self.normal = normal
		self.velocity = velocity
		self.hitEntity = hitEntity
	end
end
return {
	ProjectileCollideClientSignal = ProjectileCollideClientSignal,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/Imports/Core/Core~/src/Client/Controllers/Damage/Projectile/ProjectileCollideClientSignal.ts
import { Entity } from "Shared/Entity/Entity";
import { Projectile } from "Shared/Projectile/Projectile";

export class ProjectileCollideClientSignal {
	constructor(
		public readonly projectile: Projectile,
		public readonly hitPosition: Vector3,
		public readonly normal: Vector3,
		public readonly velocity: Vector3,
		public readonly hitEntity: Entity | undefined,
	) {}
}
